apps/backend/src/app/modules/admin/admin.service.ts

```diff
@@ -270,11 +270,23 @@ export class AdminService {
       activities.push(AdminActivityType.USER_UPDATE_ALIAS);
     }
 
-    if (update.bio && user.profile.bio !== update.bio) {
+    if (update.bio !== undefined && user.profile.bio !== update.bio) {
       updateInput.profile = { update: { bio: update.bio } };
       activities.push(AdminActivityType.USER_UPDATE_BIO);
     }
 
+    if (update.socials) {
+      for (const [k, v] of Object.entries(update.socials)) {
+        if (v === '') delete update.socials[k];
+      }
+
+      if (updateInput.profile?.update) {
+        updateInput.profile.update.socials = update.socials;
+      } else {
+        updateInput.profile = { update: { socials: update.socials } };
+      }
+    }
+
```

Comment:
This code is too complex to duplicate. I'd rather we do additional DI injecting UserService into here than have this in two different places.